optuna/terminator/improvement/evaluator.py

```diff
@@ -59,41 +68,55 @@ def get_preprocessing(self, add_random_inputs: bool = False) -> BasePreprocessin
             ),
             UnscaleLog(),
             ToMinimize(),
+            OneToHot(),
         ]
 
-        if add_random_inputs:
-            processes += [AddRandomInputs(self._min_lcb_n_additional_samples)]
-
-        processes += [OneToHot()]
-
         return PreprocessingPipeline(processes)
 
     def evaluate(
         self,
-        trials: List[FrozenTrial],
+        trials: list[FrozenTrial],
         study_direction: StudyDirection,
     ) -> float:
         search_space = IntersectionSearchSpace().calculate(trials)
         self._validate_input(trials, search_space)
 
         fit_trials = self.get_preprocessing().apply(trials, study_direction)
-        lcb_trials = self.get_preprocessing(add_random_inputs=True).apply(trials, study_direction)
 
-        n_params = len(search_space)
-        n_trials = len(fit_trials)
+        x, bounds = _convert_trials_to_tensors(trials)
+
+        y = torch.tensor([trial.value for trial in trials], dtype=torch.float64)
+        y = torch.unsqueeze(y, 1)
 
-        self._gp.fit(fit_trials)
+        gp = SingleTaskGP(
+            x,
+            y,
+            input_transform=Normalize(d=x.shape[1], bounds=bounds),
+            outcome_transform=Standardize(m=1),
+        )
 
-        ucb = _min_ucb(trials=fit_trials, gp=self._gp, n_params=n_params, n_trials=n_trials)
-        lcb = _min_lcb(trials=lcb_trials, gp=self._gp, n_params=n_params, n_trials=n_trials)
+        mll = gpytorch.mlls.ExactMarginalLogLikelihood(gp.likelihood, gp)
 
-        regret_bound = ucb - lcb
+        fit_gpytorch_model(mll)
 
-        return regret_bound
+        beta = _get_beta(n_params=len(search_space), n_trials=len(fit_trials))
+        neg_lcb_func = UpperConfidenceBound(gp, beta=beta, maximize=False)
```

Comment:
The negated LCB function and the UCB function are not same, I think. I mean
`- (\mu(x) - \beta \sigma(x)) != \mu(x) + \beta \sigma(x)`

Reply:
But we set `maximize=False`, which negates \mu(x). So, in all, this is actually the correct LCB.

Reply:
I think it's a valid argument that this is not very readable. Do you think adding 20- lines of code to define our own `LowerConfidenceBound` is preferrable?

Reply:
I see. I misunderstood the minimization and maximization. Looks reasonable.